src/libcore/option.rs

```diff
@@ -1064,6 +1064,12 @@ impl<T> From<T> for Option<T> {
     }
 }
 
+impl<'a, T> From<&'a Option<T>> for Option<&'a T> {
```

Comment:
> I'm not sure if any annotations regarding the stabilization are needed

Yes this needs to be `#[stable(feature = "option_ref_from_ref_option", since = "1.30.0")]`. (The feature name can be anything unique). This is going to be insta-stable.